client/src/components/guard/UserDataProvider.vue

```diff
@@ -0,0 +1,27 @@
+<template>
+    <div>
+        <RouterView></RouterView>
+    </div>
+</template>
+
+<script>
+import authService from '@/services/auth/authService';
+import store from '@/store';
+
+export default {
+    name: 'UserDataProvider',
+    async beforeRouteEnter(to, from, next) {
+        if (!store.state.auth.user && authService.getToken()) {
+            try {
+                await store.dispatch('auth/fetchLoggedUser');
+                next({ path: to });
+            } catch (error) {
+                console.log(error.message);
```

Comment:
for now use console.error(error);

but it must be changed later, create a task for this